src/mongocxx/collection.cpp

```diff
@@ -91,7 +91,7 @@ mongocxx::stdx::optional<bsoncxx::document::value> find_and_modify(
         if (!wc->is_acknowledged() && opts["collation"]) {
             throw mongocxx::logic_error{mongocxx::error_code::k_invalid_parameter};
         }
-        opts_builder.append(concatenate(wc->to_document()));
+        opts_builder.append(kvp("writeConcern", wc->to_document()));
```

Comment:
This is the fix. This differs from https://github.com/mongodb/mongo-cxx-driver/commit/cf4faedfa96c2b63e846a0ec169d458dfc84dbd4 since this code changed in 3.3.